App.ts

```diff
@@ -44,62 +49,56 @@ class App {
     router.get('/app/restaurants/', (req, res) => {
       console.log('Query All Restaurants');
       this.Restaurants.retrieveAllRestaurants(res);
-  });
-  router.get('/app/items/', (req, res) => {
-    console.log('Query All items');
-    this.Items.retrieveAllItems(res);
-});
-
-    // TODO: delete this
-    // router.get('/app/list/:listId/count', (req, res) => {
-    //     var id = req.params.listId;
-    //     console.log('Query single list with id: ' + id);
-    //     this.Tasks.retrieveTasksCount(res, {listId: id});
-    // });
-
-    // router.post('/app/list/', (req, res) => {
-    //   const id = crypto.randomBytes(16).toString("hex");
-    //   console.log(req.body);
-    //     var jsonObj = req.body;
-    //     jsonObj.listId = id;
-    //     this.Lists.model.create([jsonObj], (err) => {
-    //         if (err) {
-    //             console.log('object creation failed');
-    //         }
-    //     });
-    //     res.send('{"id":"' + id + '"}');
-    // });
-
-    // router.post('/app/list2/', (req, res) => {
-    //   const id = crypto.randomBytes(16).toString("hex");
-    //   console.log(req.body);
-    //     var jsonObj = req.body;
-    //     jsonObj.listId = id;
-    //     let doc = new this.Lists.model(jsonObj);
-    //     doc.save((err) => {
-    //        console.log('object creation failed');
-    //     });
-    //     res.send('{"id":"' + id + '"}');
-    // });
-
-    // router.get('/app/list/:listId', (req, res) => {
-    //     var id = req.params.listId;
-    //     console.log('Query single list with id: ' + id);
-    //     this.Tasks.retrieveTasksDetails(res, {listId: id});
-    // });
-
-    // router.get('/app/list/', (req, res) => {
-    //     console.log('Query All list');
-    //     this.Lists.retrieveAllLists(res);
-    // });
-
-    // router.get('/app/listcount', (req, res) => {
-    //   console.log('Query the number of list elements in db');
-    //   this.Lists.retrieveListCount(res);
-    // });
+    });
+    router.get('/app/items/', (req, res) => {
+      console.log('Query All items');
+      this.Items.retrieveAllItems(res);
+    });
+    router.get('/app/restaurantmanagers/', (req, res) => {
+      console.log('Query All Restaurant Managers');
+      this.RestaurantManagers.retrieveAllRestaurantManagers(res);
+    });
+    router.get('/app/restaurantowners/', (req, res) => {
+      console.log('Query All Restaurant Owners');
+      this.RestaurantOwners.retrieveAllRestaurantOwners(res);
+    });
+    router.post('/app/restaurantmanagers/create-manager', (req, res) => {
+      //TODO: add prechecks
```

Comment:
Precheck is missing. It should check the restaurantOwnerID passed is valid

Reply:
Can you explain what you mean by valid? (are you talking about the case where the restaurantOwnerID passed is already present?)

Reply:
note: we decided to only that the restaurantOwnerID field is not empty